src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -85,11 +85,11 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
         LOG.info("Get request for create new version of file={}", request.fileId)
         synchronized(this.fileBuffer) {
             val project = this.fileBuffer.getValue(request.info.projectId)
-            val patchList = project[request.fileId]?.patchesList ?: mutableListOf()
-            project[request.fileId] = CosmasProto.FileVersion.newBuilder()
-                    .setContent(request.file)
-                    .addAllPatches(patchList)
-                    .build()
+            val newVersion = project[request.fileId]?.toBuilder() ?: CosmasProto.FileVersion.newBuilder()
+            val oldText = newVersion.content.toStringUtf8()
+            val newText = request.file.toStringUtf8()
+            newVersion.addPatches(getDiffPatch(oldText, newText, this.clock.millis()))
```

Comment:
Any sense to pass timestamps from the outside as well? Because now patches and delete requests have external timestamps while createVersion uses its own. 
If you decide to supply timestamp externally, feel free to do it in a separate PR.

Reply:
Sounds reasonable, however Papeeria doesn't use this function. But I think I have enough time to fix it.